operator/pkg/controller/clusterconfig/controlplane/ingress.go

```diff
@@ -0,0 +1,273 @@
+package controlplane
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+	metalk8sscalitycomv1alpha1 "github.com/scality/metalk8s/operator/api/v1alpha1"
+	"github.com/scality/metalk8s/operator/pkg/controller/utils"
+)
+
+type IngressReconciler struct {
+	handler *utils.ObjectHandler
+	logger  logr.Logger
+
+	instance *metalk8sscalitycomv1alpha1.ClusterConfig
+}
+
+const (
+	// Name of the namespace used for the Ingress
+	namespaceName = "metalk8s-ingress"
+
+	// Name of the Pool to manage the VIP (if needed)
+	poolName = "ingress-control-plane-managed-vip"
+)
+
+// Load handler and instance in the struct
+func (r *IngressReconciler) Load(handler *utils.ObjectHandler, logger logr.Logger, instance *metalk8sscalitycomv1alpha1.ClusterConfig) {
+	r.handler = handler
+	r.logger = logger
+	r.instance = instance
+}
+
+func (r *IngressReconciler) GetLogger() logr.Logger {
+	return r.logger
+}
+
+// Reconcile the Control Plane Ingress
+func (r *IngressReconciler) Reconcile(ctx context.Context) utils.SubReconcilerResult {
+	// Reconcile the Ingress namespace
+	namespaceInstance := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}
+
+	// Create or update the namespace if needed
+	// NOTE: We treat the namespace alone has nothing else can be created if this one
+	// do not exist
+	// TODO: Use CreateOrUpdateOrDelete, It's not done yet as we do not want to Own the namespace
+	// since we not yet manage it entirely, just ensure that it exist for the moment
```

Comment:
Consider also wp ingress in the same namespace

Reply:
Fixed in https://github.com/scality/metalk8s/pull/4000/commits/827fb5dff969e89c7e4c129e69421e319871fe69